server/src/analytics.ts

```diff
@@ -0,0 +1,104 @@
+import { FastifyLoggerInstance } from "fastify";
+import { URLSearchParams } from "url";
+import dotenv from "dotenv";
+import findup from "findup-sync";
+import https from "https";
+
+export const SERVER_ID = "server";
+
+const endpointConfig = {
+  hostname: "www.google-analytics.com",
+  port: 443,
+  path: "/collect",
+  method: "POST",
+};
+
+loadEnv();
+const ANALYTICS_ID = process.env.ANALYTICS_ID;
+
+export interface AnalyticsEvent {
+  clientId: string;
+  category: string;
+  action: string;
+  label?: string;
+  value?: number;
+}
+
+export async function sendAnalytics(event: AnalyticsEvent, logInstance?: FastifyLoggerInstance): Promise<void> {
+  const promise = new Promise<void>(resolve => {
```

Comment:
rename const promise to be const request

Reply:
fixed